In [68]:
# Dependencies and Setup
import pandas as pd
import numpy as np

# File to load 
file_to_load ="../Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data_df = pd.read_csv(file_to_load)

#Print summary head() table
purchase_data_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44


Player Count:
- Display the total number of players


In [69]:
# Collecting the columns as they appear
purchase_data_df.columns

Index(['Purchase ID', 'SN', 'Age', 'Gender', 'Item ID', 'Item Name', 'Price'], dtype='object')

In [70]:
# Checking for rows of missing data
purchase_data_df.count()

Purchase ID    780
SN             780
Age            780
Gender         780
Item ID        780
Item Name      780
Price          780
dtype: int64

In [71]:
# Counting the number of players
number_of_players_count = purchase_data_df["SN"].nunique()

number_of_players_count

576

In [72]:
# Placing the number of players into a dictionary
number_of_players_dict = [{"Total Number of Players": (number_of_players_count)}]

# Storing information above in data frame
total_number_of_players_df = pd.DataFrame(number_of_players_dict)

total_number_of_players_df

Total Number of Players
0                      576

Purchasing Analysis (Total):
- Run basic calculations to obtain number of unique items, average price, etc.
- Create a summary data frame to hold the results
- Optional: give the displayed data cleaner formatting
- Display the summary data frame

In [73]:
# Using .describe() display the statistical overview of the data frame
purchase_data_df.describe()

Purchase ID         Age     Item ID       Price
count   780.000000  780.000000  780.000000  780.000000
mean    389.500000   22.714103   92.114103    3.050987
std     225.310896    6.659444   52.775943    1.169549
min       0.000000    7.000000    0.000000    1.000000
25%     194.750000   20.000000   48.000000    1.980000
50%     389.500000   22.000000   93.000000    3.150000
75%     584.250000   25.000000  139.000000    4.080000
max     779.000000   45.000000  183.000000    4.990000

In [74]:
# Counting the number of unique items
unique_items = purchase_data_df["Item ID"].nunique()
unique_items

183

In [75]:
# Counting the average price
average_price = round(purchase_data_df["Price"].mean(), 2)
average_price                      

3.05

In [76]:
# Counting number of purchases
number_purchases = purchase_data_df["Item Name"].count()
number_purchases

780

In [77]:
# Counting total revenue
total_revenue = purchase_data_df["Price"].sum()
total_revenue

2379.77

In [78]:
# Placing all the data above into a dictionary 
purchasing_analysis_dict = [{"Number of Unique Items": (unique_items),"Average Price": (average_price),
                             "Number of Purchases": (number_purchases), "Total Revenue": (total_revenue)}]

# Placing information above into a data frame
purchasing_analysis_df = pd.DataFrame(purchasing_analysis_dict)

# Placing the columns in the right order
purchasing_analysis_df = purchasing_analysis_df[["Number of Unique Items", "Average Price", "Number of Purchases",
                                                 "Total Revenue"]] 

# Use .map to format all the columns
purchasing_analysis_df["Total Revenue"] = purchasing_analysis_df["Total Revenue"].map("${:.2f}".format)

purchasing_analysis_df["Average Price"] = purchasing_analysis_df["Average Price"].map("${:.2f}".format)

purchasing_analysis_df


Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     183         $3.05                  780      $2379.77

Gender Demographics:
- Percentage and Count of Male Players
- Percentage and Count of Female Players
- Percentage and Count of Other / Non-Disclosed

In [79]:
# Print the initial data frame for refernces
purchase_data_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [80]:
# Checking unique names for Gender column
gender = purchase_data_df["Gender"].unique()
gender


array(['Male', 'Other / Non-Disclosed', 'Female'], dtype=object)

In [81]:
# Gender Demographics
gender_demographics = purchase_data_df.groupby(["Gender"])


# Counting total number for genders
total_count = gender_demographics["SN"].nunique()

# Counting percentage for each gender
percentage_cnt = (total_count/purchase_data_df["SN"].nunique()) * 100

# Placing information above into a data frame
GDs_analysis_df = pd.DataFrame({"Total Count": total_count, "Percentage of Players": percentage_cnt})

# Use .map to format all the columns
GDs_analysis_df["Percentage of Players"] = GDs_analysis_df["Percentage of Players"].map("{:.2f}%".format)

# Placing columns in order
GDs_analysis_df = GDs_analysis_df[["Total Count", "Percentage of Players"]]

# Sorting data from highest to lowest value
GDs_analysis_df = GDs_analysis_df.sort_values(by = "Total Count", ascending = False)
GDs_analysis_df


Total Count Percentage of Players
Gender                                                  
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%

Purchasing Analysis (Gender):
- Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender
- Create a summary data frame to hold the results
- Optional: give the displayed data cleaner formatting
- Display the summary data frame

In [82]:
# Purchasing Analysis ( Gender)
# Creating data frame for gender analysis
gender_df = purchase_data_df.groupby(["Gender"])
gender_df.head()

Purchase ID             SN  Age                 Gender  Item ID  \
0              0        Lisim78   20                   Male      108   
1              1    Lisovynya38   40                   Male      143   
2              2     Ithergue48   24                   Male       92   
3              3  Chamassasya86   24                   Male      100   
4              4      Iskosia90   23                   Male      131   
9              9    Chanosian48   35  Other / Non-Disclosed      136   
15            15      Lisassa64   21                 Female       98   
18            18      Reunasu60   22                 Female       82   
22            22   Siarithria38   38  Other / Non-Disclosed       24   
38            38       Reulae52   10                 Female      116   
41            41      Assosia88   20                 Female        7   
55            55      Phaelap26   25                 Female       84   
82            82     Haerithp41   16  Other / Non-Disclosed      160   
111          111       Sundim98   21  Other / Non-Disclosed       41   
228          228    Jiskirran77   20  Other / Non-Disclosed       80   

                                     Item Name  Price  
0    Extraction, Quickblade Of Trembling Hands   3.53  
1                            Frenzied Scimitar   1.56  
2                                 Final Critic   4.88  
3                                  Blindscythe   3.27  
4                                         Fury   1.44  
9                 Ghastly Adamantite Protector   3.58  
15                 Deadline, Voice Of Subtlety   2.89  
18                                     Nirvana   4.90  
22                               Warped Fetish   3.81  
38                     Renewed Skeletal Katana   4.18  
41                Thorn, Satchel of Dark Souls   1.33  
55                                  Arcane Gem   3.79  
82                                  Azurewrath   4.40  
111                                      Orbit   4.75  
228                                  Dreamsong   3.39

In [83]:
# Calculating the total number for genders
total_count = gender_demographics["SN"].nunique()
total_count

Gender
Female                    81
Male                     484
Other / Non-Disclosed     11
Name: SN, dtype: int64

In [84]:
# Calculating purchase count 
purchase_count = gender_df["Purchase ID"].count()
purchase_count

Gender
Female                   113
Male                     652
Other / Non-Disclosed     15
Name: Purchase ID, dtype: int64

In [85]:
# Counting the average price
price_avg = round(gender_df["Price"].mean(), 2)
price_avg


Gender
Female                   3.20
Male                     3.02
Other / Non-Disclosed    3.35
Name: Price, dtype: float64

In [86]:
# Calculating the total purchase
total_purchases = round(gender_df["Price"].sum(), 2)
total_purchases

Gender
Female                    361.94
Male                     1967.64
Other / Non-Disclosed      50.19
Name: Price, dtype: float64

In [87]:
# Calculating the total avg.total per person
avg_total = gender_df["Price"].sum() / gender_demographics["SN"].nunique()
avg_total

Gender
Female                   4.468395
Male                     4.065372
Other / Non-Disclosed    4.562727
dtype: float64

In [88]:
# Placing results in the data frame
summary_df = pd.DataFrame({"Purchase Count": purchase_count, "Average Purchase Price": price_avg,
                           "Total Purchase Value": total_purchases, "Average Total Purchase per Person": avg_total})

# Use .map to format all the columns

summary_df["Total Purchase Value"] = summary_df["Total Purchase Value"].astype(float).map("${:.2f}".format)
summary_df["Average Total Purchase per Person"] = summary_df["Average Total Purchase per Person"].astype(float).map("${:.2f}".format)
summary_df["Average Purchase Price"] = summary_df["Average Purchase Price"].astype(float).map("${:.2f}".format)

summary_df



Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Average Total Purchase per Person  
Gender                                                                        
Female                             $361.94                             $4.47  
Male                              $1967.64                             $4.07  
Other / Non-Disclosed               $50.19                             $4.56

Age Demographics:
- Establish bins for ages
- Categorize the existing players using the age bins. Hint: use pd.cut()
- Calculate the numbers and percentages by age group
- Create a summary data frame to hold the results
- Optional: round the percentage column to two decimal points
- Display Age Demographics Table

In [103]:

# Create the bins in which data will be held
age_bins = [0, 9, 14, 19, 24, 29, 34, 39, 100]

# Create labels for this bins
group_labels = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"] 

# Slice the data and place it into bins
purchase_data_df["Age Range"] = pd.cut(purchase_data_df["Age"], age_bins, labels = group_labels)

age_demographics = purchase_data_df.groupby("Age Range")
# Calculating precentages and numbers by age group

total_count = age_demographics["SN"].nunique()
total_players = purchase_data_df["SN"].nunique()
percentage_of_players = (total_count/total_players) *100


# Placing information into a data frame
summary_df = pd.DataFrame({"Total Count": total_count, "Percentage of Players": percentage_of_players})


# Use .map to format all the columns

summary_df["Percentage of Players"] = summary_df["Percentage of Players"].astype(float).map("{:.2f}%".format)
summary_df


Total Count Percentage of Players
Age Range                                   
<10                 17                 2.95%
10-14               22                 3.82%
15-19              107                18.58%
20-24              258                44.79%
25-29               77                13.37%
30-34               52                 9.03%
35-39               31                 5.38%
40+                 12                 2.08%

Purchasing Analysis (Age):
- Bin the purchase_data data frame by age
- Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below
- Create a summary data frame to hold the results
- Optional: give the displayed data cleaner formatting
- Display the summary data frame

In [121]:
# Create the bins in which data will be held
age_bins = [0, 9, 14, 19, 24, 29, 34, 39, 100]

# Create labels for this bins
group_labels = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"] 

# Slice the data and place it into bins
purchase_data_df["Age Range"] = pd.cut(purchase_data_df["Age"], age_bins, labels = group_labels)
purchase_data_df.head()

# Calculating the purchase count,average purchase price, avgerage perchase total per person

total_count = purchase_data_df.groupby(["Age Range"]).sum()["Price"]
total_count

avg_price = round(purchase_data_df.groupby(["Age Range"]).mean()["Price"], 2)
avg_price

purchase_count = purchase_data_df.groupby(["Age Range"]).count()["Price"]
purchase_count

avg_per_person = round((total_count/purchase_count), 2)
avg_per_person

# Placing information into a data frame
age_purchase_df = pd.DataFrame({"Purchase Count": purchase_count, "Average Purchase Price": avg_price, 
                   "Total Purchase Value": total_count, "Average Purchase Total Per Person": avg_per_person})
age_purchase_df

# Use .map to format the columns
age_purchase_df["Average Purchase Price"] = age_purchase_df["Average Purchase Price"]. map("${:.2f}".format)
age_purchase_df["Total Purchase Value"] = age_purchase_df["Total Purchase Value"]. map("${:.2f}".format)
age_purchase_df["Average Purchase Total Per Person"] = age_purchase_df["Average Purchase Total Per Person"]. map("${:.2f}".format)
age_purchase_df

Purchase Count Average Purchase Price Total Purchase Value  \
Age Range                                                               
<10                    23                  $3.35               $77.13   
10-14                  28                  $2.96               $82.78   
15-19                 136                  $3.04              $412.89   
20-24                 365                  $3.05             $1114.06   
25-29                 101                  $2.90              $293.00   
30-34                  73                  $2.93              $214.00   
35-39                  41                  $3.60              $147.67   
40+                    13                  $2.94               $38.24   

          Average Purchase Total Per Person  
Age Range                                    
<10                                   $3.35  
10-14                                 $2.96  
15-19                                 $3.04  
20-24                                 $3.05  
25-29                                 $2.90  
30-34                                 $2.93  
35-39                                 $3.60  
40+                                   $2.94

Top Spenders:
- Run basic calculations to obtain the results in the table below
- Create a summary data frame to hold the results
- Sort the total purchase value column in descending order
- Optional: give the displayed data cleaner formatting
- Display a preview of the summary data frame

In [139]:
purchase_data_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Demographics Age Range  
0  Extraction, Quickblade Of Trembling Hands   3.53            20-24     20-24  
1                          Frenzied Scimitar   1.56              40+       40+  
2                               Final Critic   4.88            20-24     20-24  
3                                Blindscythe   3.27            20-24     20-24  
4                                       Fury   1.44            20-24     20-24

In [156]:
# Create a GroupBy object based up on "SN"
sn_purchase_df = purchase_data_df.groupby(["SN"])
sn_purchase_df.head()

# Run basic calculation
sn_purchase_count = sn_purchase_df["Purchase ID"].count()
sn_purchase_count.head()

sn_purchase_avg = round(sn_purchase_df["Price"].mean(), 2)
sn_purchase_avg.head()

sn_purchase_total = round(sn_purchase_df["Price"].sum(), 2)
sn_purchase_total.head()

# Placing the information above into a DataFrame
top_spenders_df = pd.DataFrame({"Purchase Count": sn_purchase_count, "Average Purchase Price": sn_purchase_avg, 
                                "Total Purchase Value": sn_purchase_total})
top_spenders_df.head()

# Sorting data from highest to lowest value
top_spenders_df = top_spenders_df.sort_values(by = "Total Purchase Value", ascending = False)
top_spenders_df.head()

# Use .map to format the columns
top_spenders_df["Average Purchase Price"] = top_spenders_df["Average Purchase Price"].map("${:.2f}".format)
top_spenders_df["Total Purchase Value"] = top_spenders_df["Total Purchase Value"].map("${:.2f}".format)
top_spenders_df.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

Most Popular Items:
- Retrieve the Item ID, Item Name, and Item Price columns
- Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value
- Create a summary data frame to hold the results
- Sort the purchase count column in descending order
- Optional: give the displayed data cleaner formatting
- Display a preview of the summary data frame

In [157]:
purchase_data_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Demographics Age Range  
0  Extraction, Quickblade Of Trembling Hands   3.53            20-24     20-24  
1                          Frenzied Scimitar   1.56              40+       40+  
2                               Final Critic   4.88            20-24     20-24  
3                                Blindscythe   3.27            20-24     20-24  
4                                       Fury   1.44            20-24     20-24

In [179]:
# Create a GroupBy object
popular_items_df = purchase_data_df.groupby(["Item ID", "Item Name"])

item_purchase_count = popular_items_df["Purchase ID"].count()
item_purchase_count.head()

item_purchase_avg = popular_items_df["Price"].mean()
item_purchase_avg.head()

item_purchase_total = popular_items_df["Price"].sum()
item_purchase_total.head()

# Placing information above into a Data Frame
popular_items_df = pd.DataFrame({"Purchase Count": item_purchase_count, "Item Price": item_purchase_avg,
                                 "Total Purchase Value": item_purchase_total})
popular_items_df.head()


# Use .map to format the columns
popular_items_df["Item Price"] = popular_items_df["Item Price"].map("${:.2f}".format)
popular_items_df["Total Purchase Value"] = popular_items_df["Total Purchase Value"].map("${:.2f}".format)
popular_items_df.head()

# Sorting data from highest to lowest value
popular_items_df = popular_items_df.sort_values(by = "Purchase Count", ascending = False)
popular_items_df.head()


Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
82      Nirvana                                           $4.90   
19      Pursuit, Cudgel of Necromancy                     $1.02   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
108     Extraction, Quickblade Of Trembling Hands                  $31.77  
82      Nirvana                                                    $44.10  
19      Pursuit, Cudgel of Necromancy                               $8.16

Most Profitable Items:
- Sort the above table by total purchase value in descending order
- Optional: give the displayed data cleaner formatting
- Display a preview of the data frame

In [192]:
# Create a GroupBy object
popular_items_df = purchase_data_df.groupby(["Item ID", "Item Name"])

item_purchase_count = popular_items_df["Purchase ID"].count()
item_purchase_count.head()

item_price = popular_items_df["Price"].mean()
item_price.head()

item_purchase_total = item_purchase_count*item_price
item_purchase_total.head()

# Placing information above into a Data Frame
most_profitable_item_df = pd.DataFrame({"Purchase Count": item_purchase_count, "Item Price": item_price,
                                 "Total Purchase Value": item_purchase_total})
most_profitable_item_df.head()

# Sorting data from highest to lowest value
most_profitable_item_df = most_profitable_item_df.sort_values(by = "Total Purchase Value", ascending = False)
most_profitable_item_df.head()


# Use .map to format the columns
most_profitable_item_df["Item Price"] = most_profitable_item_df["Item Price"].map("${:.2f}".format)
most_profitable_item_df["Total Purchase Value"] = most_profitable_item_df["Total Purchase Value"].map("${:.2f}".format)
most_profitable_item_df.head()


Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
92      Final Critic                                               8   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
92      Final Critic                                      $4.88   
103     Singed Scalpel                                    $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
92      Final Critic                                               $39.04  
103     Singed Scalpel                                             $34.80

Heroes Of Pymoli Data Analysis:
- By looking on the “gender data” most players are male (84,03%). Also, the male group is the one who spend the most to purchase the items for the game ($ 1967.64).
- The pick demographic falls between age 20-24 years old (44.79%) the second group age is 15-19 years old (18.58%).
- The most popular and profitable item is “Oathbreaker, Last Hope of the Breaking Storm”.
